pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authentication/AuthenticationService.java

```diff
@@ -107,6 +108,8 @@ public String authenticateHttpRequest(HttpServletRequest request, Authentication
                     LOG.debug("Authentication failed for provider " + providerToUse.getAuthMethodName() + " : "
                             + e.getMessage(), e);
                 }
+                // Store the exception so we can throw it later instead of a generic one
+                authenticationException = e;
                 throw e;
```

Comment:
Notice here `e` is still thrown. So assigning `e` to `authenticationException` is meaningless because it cannot go to the logic later:

```java
            // it never reaches here because when authenticationException is not null,
            // the exception is still thrown and here it will be skipped
            if (authenticationException != null) {
                throw authenticationException;
```

Reply:
Maybe the original purpose is removing `throw e`? But the difference is only that the following check will be performed even if the authentication failed.

```java
            if (StringUtils.isNotBlank(anonymousUserRole)) {
                return anonymousUserRole;
            }
```

Reply:
Thanks for your review @BewareMyPower. I misread the original code. We can close this now.